raiden-dapp/src/components/AddressInput.vue

```diff
@@ -100,41 +117,186 @@ export default class AddressInput extends Mixins(BlockieMixin) {
     // as the input being invalid. Since the :rules prop does not support
     // async rules we have to workaround with a reactive prop
     const isAddressValid =
-      !this.busy && this.errorMessages.length === 0 && this.isAddressAvailable;
+      !this.busy &&
+      !this.typing &&
+      this.errorMessages.length === 0 &&
+      this.isAddressAvailable;
+
     if (isAddressValid) {
       this.input(this.address);
     }
 
     return [() => isAddressValid || ''];
   }
 
+  private checkAvailability(
+    value: ValidationResult
+  ): Observable<ValidationResult> {
+    return iif(
+      () => 'error' in value,
+      of(value),
+      of(value).pipe(
+        tap(() => (this.busy = true)),
+        switchMap(value => {
+          return iif(
+            () => value.value in this.presences,
+            of(this.presences[value.value]),
+            defer(() => from(this.$raiden.getAvailability(value.value)))
+          );
+        }),
+        tap(available => {
+          this.busy = false;
```

Comment:
Caution with these on/off patterns inside the pipeline. It's not that critical here, but you have async operations between them, you could end up with a value stuck at `true` because of an exception, or some funny behaviors if somehow parallel runs interleave the statements. Some patterns with increasing/decreasing integers and/or `finalize`/`using` operator combinations could be safer. Not crucial here, as the side-effect is just some spinner, but it's something important to keep in mind in this kind of scenario.